src/components/GoalsComponents/MyGoalActions/MyGoalActions.tsx

```diff
@@ -102,52 +109,88 @@ const MyGoalActions = ({ goal, open }: { open: boolean; goal: GoalItem }) => {
       </div>
       <div style={{ display: "grid", gridTemplateColumns: "1fr 1fr" }}>
         {confirmationAction && <ConfirmationModal action={confirmationAction} handleClick={handleActionClick} />}
-        <div
-          className="goal-action shareOptions-btn"
-          onClickCapture={async (e) => {
-            e.stopPropagation();
-            await openConfirmationPopUp({ actionCategory: confirmActionCategory, actionName: "delete" });
-          }}
-        >
-          <ActionDiv label={t("Delete")} icon="Delete" />
-        </div>
-        {!isPartnerGoal && (
-          <div
-            className="goal-action shareOptions-btn"
-            onClickCapture={async (e) => {
-              e.stopPropagation();
-              await openConfirmationPopUp({ actionCategory: confirmActionCategory, actionName: "archive" });
-            }}
-          >
-            <ActionDiv label={t("Done")} icon="Correct" />
-          </div>
-        )}
-        {((isPartnerGoal && goal.parentGoalId === "root") || !isPartnerGoal) && (
-          <div
-            className="goal-action shareOptions-btn"
-            onClickCapture={async (e) => {
-              e.stopPropagation();
-              if (!isPartnerGoal) {
-                handleShareGoal(goal);
-              } else {
-                await openConfirmationPopUp({ actionCategory: "collaboration", actionName: "colabRequest" });
-              }
-            }}
-          >
-            <ActionDiv
-              label={t(isPartnerGoal ? "Collaborate" : "Share")}
-              icon={isPartnerGoal ? "Collaborate" : "SingleAvatar"}
-            />
-          </div>
+        {isGoalArchived === "true" ? (
+          <>
+            <div
+              className="goal-action-archive shareOptions-btn"
+              onClickCapture={async (e) => {
+                e.stopPropagation();
+                await handleActionClick("restore");
+              }}
+            >
+              <ActionDiv
+                label={t("Restore")}
+                icon={
+                  <img
```

Comment:
pass the string not the icon. Checkout the icon component in common folder

Reply:
In the common folder there is no unarchieved icon. previously, also used same like this for the unarchived icon. So, that I pass the icon here also. can you please check this again.
![Screenshot 2023-12-10 161348](https://github.com/tijlleenders/ZinZen/assets/57442699/45050cf9-323d-4491-aac5-0a32169c543f)

Reply:
i see. no prob. thanks for trying we will resolve this in near future